internal/operators/lvm/lvm_operator.go

```diff
@@ -118,14 +120,36 @@ func (o *operator) ValidateHost(ctx context.Context, cluster *common.Cluster, ho
 		return api.ValidationResult{Status: api.Failure, ValidationId: o.GetHostValidationID(), Reasons: []string{message}}, err
 	}
 
-	// GetValidDiskCount counts the total number of valid disks in each host and return an error if we don't have the disk of required size
 	diskCount := o.getValidDiskCount(inventory.Disks, host.InstallationDiskID)
+
+	role := common.GetEffectiveRole(host)
+	message := "Logical Volume Manager requires at least one non-installation HDD/SSD disk on the host"
+	if (role == models.HostRoleWorker && !*cluster.SchedulableMasters) || *cluster.SchedulableMasters {
+		if diskCount == 0 {
+			return api.ValidationResult{Status: api.Failure, ValidationId: o.GetHostValidationID(), Reasons: []string{message}}, nil
+		}
+	}
+
+	if role == models.HostRoleAutoAssign && !*cluster.SchedulableMasters {
+		status := "For LVM Standard Mode, host role must be assigned to master or worker."
+		return api.ValidationResult{Status: api.Failure, ValidationId: o.GetHostValidationID(), Reasons: []string{status}}, nil
+	}
+
+	requirements, err := o.GetHostRequirements(ctx, cluster, host)
 	if err != nil {
-		return api.ValidationResult{Status: api.Failure, ValidationId: o.GetHostValidationID(), Reasons: []string{err.Error()}}, nil
+		message := fmt.Sprintf("Failed to get host requirements for host %s", host.ID)
+		o.log.Error(message)
```

Comment:
we're kind of losing `err` here in our logs, no?
in case it has any valuable information about the specifics of the error

Reply:
the code is changed, there is no longer an error here